src/nvim/api/window.c

```diff
@@ -93,6 +93,9 @@ void nvim_win_set_cursor(Window window, ArrayOf(Integer, 2) pos, Error *err)
   // When column is out of range silently correct it.
   check_cursor_col_win(win);
 
+  // Make sure we stick in this column.
+  win->w_curswant = win->w_cursor.col;
```

Comment:
why not use  `col` here directly? If we set to column 20 a line with 10 columns, and then `j` to a line with 20 columns, is it not more reasonable to then use column 20?

Reply:
Sure, I'm fine with that.  (Actually that was my first draft, and then I changed my mind - but for no very good reason).  I'll change it.